addons/web_editor/static/src/js/editor/odoo-editor/test/spec/utils.test.js

```diff
@@ -1541,8 +1542,143 @@ describe('Utils', () => {
+            const abc = p1.firstChild;
+            const def = p2.firstChild;
+            let [node, offset] = getDeepestPosition(editable, 0);
+            window.chai
+                .expect([node, offset])
+                .to.eql([abc, 0]);
+            [node, offset] = getDeepestPosition(editable, 1);
+            window.chai
+                .expect([node, offset])
+                .to.eql([def, 0]);
+            [node, offset] = getDeepestPosition(editable, 2);
+            window.chai
+                .expect([node, offset])
+                .to.eql([def, 3]);
+        });
+        it('should get deepest position for node with invisible element', () => {
+            const [p1] = insertTestHtml(
+                `<p></p><p>def</p>`,
+            );
+            const editable = p1.parentElement;
+            const def = editable.lastChild.firstChild;
+            let [node, offset] = getDeepestPosition(editable, 0);
+            window.chai
+                .expect([node, offset])
+                .to.eql([def, 0]);
+            [node, offset] = getDeepestPosition(editable, 2);
+            window.chai
+                .expect([node, offset])
+                .to.eql([def, 3]);
+        });
+        it('should get deepest position for invisible block element', () => {
+            const [p] = insertTestHtml(
+                `<p></p><span>def</span>`,
+            );
+            const [node, offset] = getDeepestPosition(p, 0);
+            window.chai
+                .expect([node, offset])
+                .to.eql([p, 0]);
+        });
+        it('should get deepest position for invisible block element(2)', () => {
+            const [span] = insertTestHtml(
+                `<span>abc</span><p></p>`,
+            );
+            const p = span.nextSibling;
+            const [node, offset] = getDeepestPosition(p, 0);
+            window.chai
+                .expect([node, offset])
+                .to.eql([p, 0]);
+        });
+        it('should get deepest position for banner element', () => {
+            const [banner] = insertTestHtml(
+                `<div class="o_editor_banner o_not_editable lh-1 d-flex
+                        align-items-center alert alert-info pb-0 pt-3"
+                        role="status" data-oe-protected="true" contenteditable="false">
+                    <i class="fs-4 fa fa-info-circle mb-3" aria-label="Banner Info"></i>
+                    <div class="w-100 ms-3" data-oe-protected="false" contenteditable="true">
+                        <p>abc</p>
+                        <p>def</p>
+                    </div>
+                </div>`,
+            );
+            const editable = banner.parentElement;
+            const icon = banner.childNodes[1];
+            const def = banner.childNodes[3].childNodes[3].firstChild;
+            let [node, offset] = getDeepestPosition(editable, 0);
+            window.chai
+                .expect([node, offset])
+                .to.eql([icon, 0]);
```

Comment:
Just taking the `getDeepestPosition` in account it seems correct to me but then looking at `_fixFontAwesomeSelection` I think we should not test `getDeepestPosition` for icon.

Reply:
Just taking `getDeepestPosition` in account it is correct because icon `isVisible` element but then looking at `_fixFontAwesomeSelection`. I think we should not test `getDeepestPosition` for icons.

Reply:
But the thing here is that we don't really test it for icon, we test it for editable 0, and when we do so and that the icon is the first element, the returned result is wrong IMO. It should be abc 0 in that particular case IMHO and this test would actually be very useful to ensure that we don't set the cursor in the font awesome. Now with that being said, shouldn't it be corrected by `_fixFontAwesomeSelection` ? Maybe we should wait for `nextTick` to have `selectionchange` be triggered?